os/arch/arm/src/imxrt/imxrt_allocateheap.c

```diff
@@ -363,26 +363,42 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
 #endif
 
 /****************************************************************************
- * Name: up_addregion
- *
- * Description:
- *   Memory may be added in non-contiguous chunks.  Additional chunks are
- *   added by calling this function.
- *
+ * Name: up_add_kregion
  ****************************************************************************/
+#if defined(CONFIG_MM_KERNEL_HEAP) && (CONFIG_KMM_REGIONS > 1)
+static void up_add_kregion(void)
+{
+	int region_cnt;
+	struct mm_heap_s *kheap;
+	kheap = kmm_get_heap();
+	for (region_cnt = 1; region_cnt < CONFIG_KMM_REGIONS; region_cnt++) {
+		if (kheap[regionx_kheap_idx[region_cnt]].mm_heapsize == 0) {
+			mm_initialize(&kheap[regionx_kheap_idx[region_cnt]], kregionx_start[region_cnt], kregionx_size[region_cnt]);
+			continue;
+		}
+		mm_addregion(&kheap[regionx_kheap_idx[region_cnt]], kregionx_start[region_cnt], kregionx_size[region_cnt]);
+	}
+}
+#endif
 
-#if CONFIG_MM_REGIONS > 1
+/****************************************************************************
+ * Name: up_addregion
+ ****************************************************************************/
+#if (CONFIG_MM_REGIONS > 1) || (defined(CONFIG_MM_KERNEL_HEAP) && (CONFIG_KMM_REGIONS > 1))
 void up_addregion(void)
```

Comment:
Why I remain this function as now, we can give option to developers which function to use when allocating heap. 
They can use/select their own logic, or our common logic which is in os/arch/arm/src/common/up_allocateheap.c
What do you think about this?

Reply:
The APIs for allocating the user and kernel heaps is slightly different in imxrt from the common code. So we cannot use common/up_allocateheap.c for imxrt. That is why I added these APIs from the common code to the imxrt file.

Reply:
I'm not sure what is the difference between common code and this new.

Reply:
In imxrt, since we have separate sdram region, we might want to use it either for the user heap or the kernel heap. This logic is part of up_allocate_heap and up_allocate_kheap APIs in imxrt. So, the imxrt allocateheap.c needs to be different from the up_allocateheap.c.